x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/RegressionInferenceResults.java

```diff
@@ -19,51 +19,49 @@
 
     public static final String NAME = "regression";
 
-    public RegressionInferenceResults(double value) {
+    private final String resultsField;
+
+    public RegressionInferenceResults(double value, InferenceConfig config) {
         super(value);
+        assert config instanceof RegressionConfig;
+        RegressionConfig regressionConfig = (RegressionConfig)config;
+        this.resultsField = regressionConfig.getResultsField();
     }
 
     public RegressionInferenceResults(StreamInput in) throws IOException {
         super(in.readDouble());
+        this.resultsField = in.readString();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-    }
-
-    @Override
-    XContentBuilder innerToXContent(XContentBuilder builder, Params params) {
-        return builder;
+        out.writeString(resultsField);
     }
 
     @Override
     public boolean equals(Object object) {
         if (object == this) { return true; }
         if (object == null || getClass() != object.getClass()) { return false; }
         RegressionInferenceResults that = (RegressionInferenceResults) object;
-        return Objects.equals(value(), that.value());
+        return Objects.equals(value(), that.value()) && Objects.equals(this.resultsField, that.resultsField);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(value());
+        return Objects.hash(value(), resultsField);
     }
 
     @Override
-    public void writeResult(IngestDocument document, String resultField, InferenceConfig config) {
+    public void writeResult(IngestDocument document, String resultField) {
         ExceptionsHelper.requireNonNull(document, "document");
         ExceptionsHelper.requireNonNull(resultField, "resultField");
-        document.setFieldValue(resultField, value());
+        document.setFieldValue(resultField + "." + resultsField, value());
```

Comment:
It took me a while to see the difference between `resultField` and `result[s]Field` can we think of a better name

Reply:
👍 i will name it "parentResultField" since that is what it is. Cool?